camel/agents/task_agent.py

```diff
@@ -1,377 +1,284 @@
+                **meta_dict))
+
+        task_msg = BaseMessage.make_user_message(
+            role_name="Task Specifier", content=self.task_specify_prompt)
+        specifier_response = self.step(task_msg)
+        if (specifier_response.msgs is None
+                or len(specifier_response.msgs) == 0):
+            raise RuntimeError("Task specification failed.")
+        specified_task_msg = specifier_response.msgs[0]
+
+        if specifier_response.terminated:
+            raise RuntimeError("Task specification failed.")
+
+        return TextPrompt(specified_task_msg.content)
+
+
+class TaskPlannerAgent(ChatAgent):
+    r"""An agent that helps divide a task into subtasks based on the input
+    task prompt.
+
+    Attributes:
+        task_planner_prompt (TextPrompt): A prompt for the agent to divide
+            the task into subtasks.
+
+    Args:
+        model (ModelType): The type of model to use for the agent.
+            (default: :obj:`ModelType.GPT_3_5_TURBO`)
+        model_config (Any): The configuration for the model.
+            (default: :obj:`None`)
+        output_language (str, optional): The language to be output by the
+        agent. (default: :obj:`None`)
+    """
+
+    def __init__(
+        self,
+        model: Optional[ModelType] = None,
+        model_config: Any = None,
+        output_language: Optional[str] = None,
+    ) -> None:
+
+        self.task_planner_prompt = TextPrompt(
+            "Divide this task into subtasks: {task}. Be concise.")
+
+        system_message = BaseMessage(
+            role_name="Task Planner",
+            role_type=RoleType.ASSISTANT,
+            meta_dict=None,
+            content="You are a helpful task planner.",
+        )
+
+        super().__init__(system_message, model, model_config,
+                         output_language=output_language)
+
+    def run(
+        self,
+        task_prompt: Union[str, TextPrompt],
+    ) -> TextPrompt:
+        r"""Generate subtasks based on the input task prompt.
+
+        Args:
+            task_prompt (Union[str, TextPrompt]): The prompt for the task to
+                be divided into subtasks.
+
+        Returns:
+            TextPrompt: A prompt for the subtasks generated by the agent.
+        """
+        # TODO: Maybe include roles information.
+        self.reset()
+        self.task_planner_prompt = self.task_planner_prompt.format(
+            task=task_prompt)
+
+        task_msg = BaseMessage.make_user_message(
+            role_name="Task Planner", content=self.task_planner_prompt)
+        task_response = self.step(task_msg)
+
+        if task_response.msgs is None:
+            raise RuntimeError("Got None Subtasks messages.")
+        if task_response.terminated:
+            raise RuntimeError("Task planning failed.")
+
+        sub_tasks_msg = task_response.msgs[0]
+        return TextPrompt(sub_tasks_msg.content)
+
+
+class RoleAssignmentAgent(ChatAgent):

```

Comment:
Is this conflict with the role_assignment_agent.py? Dimitri said it's better separate the role assignment agent to a separate one